### PR TITLE
Put footnotes on the same line with `fmt_markdown()` in Quarto

### LINKED ISSUE #1773: Footnote marker placement when using `gt::md()` in Quarto in gt v0.11.0
## Prework

* [x] Read and agree to the [code of conduct](https://www.contributor-covenant.org/version/2/1/code_of_conduct.html) and [contributing guidelines](https://github.com/rstudio/gt/blob/master/.github/CONTRIBUTING.md).
* [x] If there is [already a relevant issue](https://github.com/rstudio/gt/issues), whether open or closed, comment on the existing thread instead of posting a new issue.

## Description

When a footnote is placed next to text that has been processed with `gt::md()`, the footnote marker is placed in a separate row when rendered with Quarto. I spotted this behaviour in both column labels and footnotes (I haven't checked all location types, e.g. table cells, grouped row headers, summary rows, etc FYI). Note this does not occur when running the code interactively in the RStudio IDE. I think this is new behavior in gt 0.11.0.

## Reproducible example

In the example below, I am using `gt::md()` in the column label and in the actual footnote text. In both cases, the use of `gt::md()` drives the footnote marker so its own row.

````
---
title: "Untitled"
format:
  html:
    embed-resources: true
---

## Quarto

gt footnotes without `gt::md()`: no issues.

```{r}
mtcars[1:2, 1:2] |> 
  gt::gt() |> 
  gt::cols_label(mpg = "**MPG**") |> 
  gt::tab_footnote(
    "_Adding footnote_",
    locations = gt::cells_column_labels(columns = gt::everything())
  )
```

gt footnotes **WITH** `gt::md()`: footnote markers appear on their own line.

```{r}
mtcars[1:2, 1:2] |> 
  gt::gt() |> 
  gt::cols_label(mpg = gt::md("**MPG**")) |> 
  gt::tab_footnote(
    gt::md("_Adding footnote_"),
    locations = gt::cells_column_labels(columns = gt::everything())
  )
```
````

image

Thank you again!!! 🕺🏼 

PS This issue could be related to https://github.com/rstudio/gt/issues/1541. But that doesn't mention Quarto?

### PR DESCRIPTION
**Pull request by olivroy**
# Summary

It seems that using a span instead of div seems to work!!

Sorry about the noise. I also think it is not necessary to keep the gt_from_md class?

# Related GitHub Issues and PRs

Fixes #1773 

# Checklist

- [x] I understand and agree to the [Code of Conduct](https://www.contributor-covenant.org/version/2/1/code_of_conduct.html).
- [x] I have listed any major changes in the [NEWS](https://github.com/rstudio/gt/blob/master/NEWS.md).
- [x] I have added [`testthat`](https://github.com/r-lib/testthat) unit tests to [`tests/testthat`](https://github.com/rstudio/gt/tree/master/tests/testthat) for any new functionality.

### CODE REVIEW

**Comment by olivroy** [comment]
I am using the following script to compare diff

```r
#
gt_tbl <- mtcars[1:2, 1:2] |>
  gt::gt() |>
  gt::cols_label(mpg = gt::md("**MPG**")) |>
  gt::tab_footnote(
    gt::md("_Adding footnote_"),
    locations = gt::cells_column_labels(columns = gt::everything())
  )
# inside Quarto
html_code_quarto <- withr::with_envvar(c("QUARTO_BIN_PATH" = "true"), gt:::render_as_html(gt_tbl))
# out of Quarto
html_code <- withr::with_envvar(c("QUARTO_BIN_PATH" = ""), gt:::render_as_html(gt_tbl))

waldo::compare(html_code, html_code_quarto, x_arg = "html", y_arg = "quarto")

html_file <- withr::local_tempfile(lines = html_code, pattern = "html")
quarto_file <- withr::local_tempfile(lines = html_code_quarto, pattern = "quarto")
diffviewer::visual_diff(
  html_file,
  quarto_file
)
```

image

The diff seems good. I will look if adding back the unprocessed text works well.

**Review by rich-iannone — APPROVED** [review]
LGTM!

File: R/quarto.R [hunk 1]
```diff
@@ -40,12 +40,14 @@ process_md_quarto <- function(text, md_engine_fn) {
...
 
+  # Remove paragraph
+  non_na_text_processed <- gsub("^<p>|</p>\n$", "", non_na_text_processed)
   # Use base64 encoding to avoid issues with escaping internal double
   # quotes; used in conjunction with the 'data-qmd-base64' attribute
   # that is recognized by Quarto
   non_na_text <-
     vapply(
-      non_na_text,
+      non_na_text_processed,
```
Comment (olivroy) [inline R/quarto.R:50]: This is the bad / unnecessary change and it is reverted in #1958
